src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
-    [DebuggerDisplay("Name: {Declaration.Name}, Namespace: {Declaration.Namespace}")]
```

Comment:
the old display here is using the `Declaration` property here, therefore whenever an instance of this type shows up in debugger view, this Declaration property will be called, but this is a lazy-loaded property, this debugger display changes the internal state of this instance, which is not acceptable and might cause issues.